Update the TC responsibilities w.r.t. donations

bhs opened the pull request:
This change is as discussed at a GC meeting in December. Happy to take feedback and/or discuss the proposal if the TC is concerned about feasibility. cc @open-telemetry/technical-committee @open-telemetry/governance-committee

alolita reviewed (approved):
@bhs +1 to outlining how large donations are handled by OpenTelemetry.
It may also be useful to add some detail about 
* how to contact the TC when submitting a large code donation 
* outlining requirements such as an agreement to change of copyright to reflect the OpenTelemetry Authors and code license being compatible with Apache 2.0.
* Code maintenance requirements

bhs commented:
@alolita great suggestions. I updated this PR to include more guidance (per your comment) and moved some of that guidance into `CONTRIBUTING.md` where I'd look for it if I were an outside donator.

bhs commented:
I can theoretically merge this given the approvals, but I changed the PR in meaningful ways since those approvals came through. Would love an explicit 👍 from an approver before this gets merged.

tigrannajaryan commented on CONTRIBUTING.md:
```diff
@@ -225,6 +225,27 @@ contributors, so we would like you to come join us! The [Community Membership
+guidance at the TC's earliest convenience.
+
+All donated code requires a license compatible with the Apache Software License
+2.0 (ASL 2), and donated code will require a change of copyright to reflect the
```
Nit: I have never seen abbreviation "ASL 2" before. Not sure what the purpose of using it here is.

bhs replied:
interesting – I have seen it, but I am also glad to remove it :)  Will do.
